streams/src/main/java/org/apache/kafka/streams/kstream/CogroupedKStream.java

```diff
@@ -275,6 +275,15 @@
      */
     <W extends Window> TimeWindowedCogroupedKStream<K, VOut> windowedBy(final Windows<W> windows);
 
+    /**
+     * Create a new {@link TimeWindowedCogroupedKStream} instance that can be used to perform sliding
+     * windowed aggregations.
+     *
+     * @param windows the specification of the aggregation {@link SlidingWindows}
+     * @return an instance of {@link TimeWindowedCogroupedKStream}
+     */
+    TimeWindowedCogroupedKStream<K, VOut> windowedBy(final SlidingWindows windows);
```

Comment:
This PR is rather larger. Would it maybe make sense to split it into 2 and add co-group in it's own PR?

Reply:
We could, but it would only pull out 3ish classes and not very many lines, so I don't think it would make this PR feel much smaller

Reply:
Seems like the cogroup stuff makes up a pretty small amount of the overall PR, but up to Leah

Reply:
Ack. Was just a thought.

Reply:
Coming back to this after completing the review, I'd say the biggest advice I'd share is to avoid whitespace changes and cleanups on the side when the PR is so long already. In fact, for my own super-complex PRs, I tend to go back over the whole diff and back out anything that's not critically important, just to lighten the load on the reviewers.

Cleanups are nice to have, but it's better to keep them in their own PRs or in more trivial ones.